### ISSUE TITLE
CAF::Process supports shell option

### ISSUE THREAD

**Issue by stdweird** [quattor/CAF#255]
currently, `CAF::Process` support the `LC::Process` shell `option` (ie assume there's only one element in the command list, and run it as is (incl lack of protection again shell-isms).
we should eitehr document it or remove it. i'm in favour of removing it, but it goes against the 'should allow to shoot yourself in the foot` principle (also known as `CAF devs shouldn't pretend to know it all` :wink:)

**Comment by jouvin** [comment]
LAL workshop: first step, throw a deprecation warning and at the next workshop, decide to remove the support.

**Comment by jrha** [comment]
N.B. ncm-filecopy appears to use `shell => 1` for restart commands.

**Comment by jouvin** [comment]
Exactly, at least `ncm-filecopy` makes use of it. This seems difficult to remove it, except if we tokenize the command given for the `restart` property in `filecopy`. The goal of the deprecation warning was to better identify all the circumstances where a shell is used. I'm afraid that there is a couple of grid components also rely on this feature but I don't remember precisely.

**Comment by jrha** [comment]
AFAICT it's only `ncm-glitestartup`

**Comment by jrha** [comment]
As discussed we need a migration strategy for (at least) filecopy that does not place a burden on sites when upgrading.

**Comment by jrha** [comment]
No consensus yet, hopefully we can **discuss at workshop**.

**Comment by jrha** [comment]
Discussed at workshop, an option to reduce usage of subshell in filecopy would be to allow a list (or list of lists) of command + arguments for the restart command.
